### ISSUE TITLE
[UX] Improve the help text on the Custom blocks page.

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#6531]
## Description of the bug

There are a number of issues and ideas floating around the issue queue about clarifying the difference between custom blocks and reusable custom blocks. Intersting work is actively progressing on this issue right now (https://github.com/backdrop/backdrop-issues/issues/4108).

When looking at that issue, it occurred to me that there is a small item not addressed in that PR which I think would be very easy to do and be a significant improvement as we think about and work on more significant changes/improvements. 

The Custom Blocks page in the admin Structure menu, is actually somewhat confusing. I would like to propose two very simple and specific changes and as an interim step while larger and more comprehensive ideas are being discussed. 

Change the page title of this page to: Reusable custom blocks 
Change the title of the link to: Add resuable custom block

![image](https://github.com/backdrop/backdrop-issues/assets/3144571/daa8d1e3-d66c-436a-ae4d-f029f4617741)

ORIGINAL PROPOSAL:

![image](https://github.com/backdrop/backdrop-issues/assets/3144571/450a8f54-4358-4f2a-add4-107023b29fa8)

CURRENT PROPOSAL (only changes help text):

![image](https://github.com/user-attachments/assets/78147b53-3237-45e2-a762-f18dd5004cc4)

## Steps To Reproduce

To reproduce the behavior:
1. Go to admin/structure/block

## Additional information

There are many other tickets right now addressing different ways of improving the UI to help people better understand the difference between regular custom blocks and reusable custom blocks. This ticket is focused on these two verys small and specific changes and does not preclude the many other ideas being discussion. 

Unless, I'm missing something, I think this might be a quick first step.

## CURRENT PR

As of Dec 10, 2024 - this PR simply makes some changes to the help text.

**Comment by stpaultim** [comment]
I've shared a PR. This PR makes a couple of other small changes for consistency. My biggest concern is changing this menu item, it will take a while for folks to get used to this change from "Custom blocks" to "Reusable custom blocks." 

For the menu, it might be less jarring a change if we list "Custom blocks (reusable)". While listing them as "Reusable custom blocks" in other places. 

![image](https://github.com/backdrop/backdrop-issues/assets/3144571/72d2cf76-c88f-4477-81bd-ede15a0c61f1)

**Comment by stpaultim** [comment]
Here is an alternative, that does not change the admin menu structure so much.

![image](https://github.com/backdrop/backdrop-issues/assets/3144571/ddc2df4c-27c0-42f3-8410-8878e902932a)

**Comment by klonos** [comment]
I'm in favor of this change here 👍🏼 

As I mentioned in #5922:
> - If we were introducing this feature now, I would certainly find `Reusable custom blocks` over the current `Custom blocks` way more fitting. So 👍🏼 from me on the change proposed

...but also:
> - Since the feature has been around for a long time, people are used to the position of the current "Custom blocks" menu item in the "Structure" top-level menu in the admin bar. Adding the word "reusable" as the first word causes the menu to be repositioned, and I expect people to find it rather disruptive/annoying. I then thought that perhaps `Custom reusable blocks` would be a good compromize, since it introduces the word "reusable" while at the same time retaining "custom" as the first word. That would in turn leave the order of the menu item in the admin bar unaffected.
> - `Reusable custom blocks` is much better than `Custom reusable blocks` though (more technically accurate).

**Comment by stpaultim** [comment]
@klonos - what about "Custom blocks (reusable)" as the admin menu label?

**Comment by quicksketch** [comment]
I think both `Custom blocks (reusable)` and `Reusable custom blocks` are a bit too verbose. Originally the page was simply called "Blocks" and we already changed it to "Custom blocks".

**Comment by stpaultim** [comment]
If we can't change the page title to make it more accurate, at a minimum we should look at improving that help text to make it more descriptive. 

Can we at least change this:

> "These blocks are available to be placed in any layout in the Layoutspage."

To this (or something like this?):

> "This page lists all Reusable Custom blocks that are available to be placed in any layout in the Layouts page. Updating the content of these reusable blocks will affect all layouts where they are placed!

**Comment by quicksketch** [comment]
I'm all for updating the help text :+1: 

Let's be consistent with "Reusable Custom blocks" vs "reusable custom blocks". Since these are not proper nouns (like the name of a module) I would suggest just all lowercase.

**Comment by klonos** [comment]
As a UX improvement/bug, this could get in at any point - we don't need to wait for a minor release. What we need is consensus!

>I think both `Custom blocks (reusable)` and `Reusable custom blocks` are a bit too verbose. Originally the page was simply called "Blocks" and we already changed it to "Custom blocks".

I do not share the same concern as @quicksketch with regards to the length of the words here, but I do share the concern of changing things again and again. That, in conjunction with the following things makes me hesitate:
- I feel that "reusable" is not 100% accurate to describe what these blocks are or how they behave
- I also feel that "reusable" does not do a good job at intuitively explaining the difference between "regular", non-reusable blocks vs. the "reusable" ones (not without documentation, help text in the UI and lots of additional explanation around what that word means in Backdrop specifically)
- _technically_ all blocks are "reusable"
- I find the terms "shared content block" or "mirrored content block" or "cloned content block" more accurate and straight-forward

**Comment by klonos** [comment]
...I am fine with changing the help text to say reusable custom blocks. I'm also fine with changing the page title (but not the admin menu links for now).

I would like to propose that in the help text we emphasize the word "reusable" (italics or bold).

**Comment by yorkshire-pudding** [comment]
I'm struggling to have a strong opinion on this.  I think both "reusable custom block" and "shared content block" are both good ideas. I don't think "mirrored content" or "cloned content" are appropriate as some custom blocks might only be used once and these descriptions suggest it has already been cloned or mirrored. Also "cloned" suggests a copy at a snapshot in time rather than keeping updated with changes across all copies.

**Comment by klonos** [comment]
Thanks @yorkshire-pudding 🙏🏼 ...that is excellent feedback 👍🏼

**Comment by stpaultim** [comment]
Let's call them "Quantum Blocks" because it sounds good and because the content lives in two places simotaneously at one time.

:-)

> It's one of the oddest tenets of quantum theory: a particle can be in two places at once ([link](https://www.science.org/content/article/one-quantum-physics-greatest-paradoxes-may-have-lost-its-leading-explanation#:~:text=It's%20one%20of%20the%20oddest,one%20of%20its%20two%20locations.))

**Comment by stpaultim** [comment]
@quicksketch and @klonos - I understand that this not urgent, so we don't have to rush this in. BUT, just in case, I removed the changes to the menu items and focused on some improvements to help text. The risk is very low here and I do think that even these small changes in help text are an improvement.

**Comment by yorkshire-pudding** [comment]
> Let's call them "Quantum Blocks" because it sounds good and because the content lives in two places simotaneously at one time.
> 
> :-)
> 
> > It's one of the oddest tenets of quantum theory: a particle can be in two places at once ([link](https://www.science.org/content/article/one-quantum-physics-greatest-paradoxes-may-have-lost-its-leading-explanation#:~:text=It's%20one%20of%20the%20oddest,one%20of%20its%20two%20locations.))

Isn't another bit of quantum theory, in quantum computing, that they can be simultaneously 0 and 1, so the blocks could be both enabled and disabled at the same time?

**Comment by klonos** [comment]
- Dr. Strange blocks 🤣 
- omnipresent blocks (or "omni-blocks" for short)

**Comment by klonos** [comment]
...in all seriousness though, I'm having a look at the PR @stpaultim 👀

**Comment by klonos** [comment]
@stpaultim I like where this is going, thank you 🙏🏼 

I've left a couple of suggestions in the PR for consideration.

Here's another suggestion:
- remove the bold `Updating the content of these reusable blocks will affect all layouts where they are placed!` text from `admin/structure/block` (lets eave that for the edit forms/dialogs when creating/editing reusable blocks, to be done as part of )
- replace that text with a variation that incorporates some of the description text that you added for the blocks, which I really like: `Create reusable custom blocks that can be placed in multiple layouts and edited in one place.`

So the final text in `admin/structure/block` could be something like this:
> This page lists all blocks with _reusable content_. Such blocks can be placed in multiple [layouts]() and editing their content can be done from a single place.

**Comment by klonos** [comment]
@jenlampton @yorkshire-pudding @olafgrabienski (and others) what do you think re the above ^^?

**Comment by klonos** [comment]
...this brainstorming makes me want to change the "Make this block reusable" label in the checkbox to "Share this content across multiple layouts". Makes it so much more straight-forward.

**Comment by yorkshire-pudding** [comment]
@klonos - I like those suggestions. Preference for "These" over "Such"

**Comment by stpaultim** [comment]
Modified with some, but not all, of the suggestions from @klonos. Sorry, I don't like "reusable content." :-(
@yorkshire-pudding - this version does not use "these" or "such."

![image](https://github.com/backdrop/backdrop-issues/assets/3144571/e072fac3-ef02-4220-9de5-6ed0afd98827)

**Comment by yorkshire-pudding** [comment]
Looks good @stpaultim

**Comment by olafgrabienski** [comment]
Same here, I liked the suggestions, and the modification by @stpaultim looks good.

**Comment by stpaultim** [comment]
This issue has been discussed. It has a PR and some emerging consensus. 

Does anyone else have any additional thoughts?

I add the Milestone Candidate tag. Anyone want to add this to the bugfix milestone?
https://docs.backdropcms.org/documentation/adding-milestones-to-issues

**Comment by stpaultim** [comment]
This issue becomes mute if https://github.com/backdrop/backdrop-issues/issues/6781 proceeds.

**Comment by quicksketch** [comment]
Yep, and in the end I ate my words since we ended up calling them "Reusable custom blocks" in the admin bar after all.

**Comment by jenlampton** [comment]
I'm going to re-open this issue because the "CURRENT PROPOSAL" In this issue is not what got in in https://github.com/backdrop/backdrop-issues/issues/6781. It looks like that was previously discussed here also, but iterated on until the "CURRENT PROPOSAL" so I'd like to see if we can get these additional changes in as UX improvements before the 1.30.0 release.

**Comment by stpaultim** [comment]
Good point. The original ideas suggested here did get in, but some of the ideas that came later were not part of that other PR, but would compliment that other PR nicely. Thanks @jenlampton

**Comment by quicksketch** [comment]
I updated the PR at https://github.com/backdrop/backdrop/pull/4989 to fix the test coverage (there is an unrelated code style error that can be ignored). This is ready for review now.

**Comment by quicksketch** [comment]
Merged https://github.com/backdrop/backdrop/pull/4989 into 1.x for 1.30.0.
